### ISSUE TITLE
deepspeed stage 3 do not save the lr_scheduler

### ISSUE THREAD

**Issue by jiangix-paper** [microsoft/DeepSpeed#3875]
Hello, I found the deepspeed stage 3 do not save the lr scheduler. So when I resume the checkpoint from past checkpoint, the learning rate is not continue.

**Comment by zyfedward** [comment]
hit same issue in deepspeed stage2

**Comment by tjruwase** [comment]
@jiangix-paper and @zyfedward thanks for reporting this issue. Can you please confirm whether you pass lr_scheduler to `deepspeed.intialize()`?

**Comment by zyfedward** [comment]
Thanks @tjruwase for helping to investigate. I can workaround the issue by using DS scheduler instead of HF scheduler. Looks like the issue is in accelerator side.

**Comment by tjruwase** [comment]
@jiangix-paper, can you please share repro details?

**Comment by tongjingqi** [comment]
Train 20 steps:
![图片](https://github.com/microsoft/DeepSpeed/assets/66310692/db135f0c-caec-4059-9fb9-ccb7d59d7b31)

Restart from 10 step
![图片](https://github.com/microsoft/DeepSpeed/assets/66310692/a476bcc6-38cb-4ed4-8202-9ee5de9b359e)

We can see both lr are same.
